### PR TITLE
[plugin.video.fullycharged] v1.0.1

### PR DESCRIPTION
**Pull request by zag2me**
### Description
Initial upload of the Fullycharged video Add-on

icon

Submitted to the krypton branch as it uses the new asset tags in the addon.xml. Hope this is the correct way of doing it?

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

**Comment by bickyz** [comment]
hi, thank you for the addon. 

I am using your addon to list all the videos from my youtube channel. Currently the addon has just one thumbnail icon that on click will show all the videos from my youtube channel.

I would like to add one more thumbnail icon that will play the live tv stream (rtmp://), how can I do this on the same video addon ? 

Any help would be much appreciated, thank you.

**Comment by zag2me** [comment]
Not the place to discuss this sorry, this is a repo pull for the official repository.

Please create an issue on the original source.

**Comment by phil65** [comment]
krypton branch is not really needed in case you move icon and fanart to the addon root. That way you could also send it to jarvis branch. Correct, @learningit ?

File: plugin.video.fullycharged/addon.xml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="plugin.video.fullycharged"
+	name="Fully Charged"
+	version="1.0.1"
+	provider-name="zag">
+  <requires>
+    <import addon="xbmc.python" version="2.1.0"/>
```
Comment (learningit) [inline plugin.video.fullycharged/addon.xml]: Further to @phil65 's comment. You need to set xbmc.python to the correct minimum branch (see http://kodi.wiki/view/Addon.xml#Dependency_versions for detail).
I would move the icon.png and fanart.jpg files into the addon root directory as well.

File: plugin.video.fullycharged/plugintools.py [hunk 2]
```diff
@@ -0,0 +1,582 @@
...
+            view_code = view_codes.get(skin_name)
+            _log("set_view view code for "+view_mode+" in "+skin_name+" is "+str(view_code))
+            xbmc.executebuiltin("Container.SetViewMode("+str(view_code)+")")
+        else:
+            _log("set_view view code forced to "+str(view_code))
+            xbmc.executebuiltin("Container.SetViewMode("+str(view_code)+")")
+    except:
+        _log("Unable to find view code for view mode "+str(view_mode)+" and skin "+skin_name)
+
+f = open( os.path.join( os.path.dirname(__file__) , "addon.xml") )
```
Comment (learningit) [inline plugin.video.fullycharged/plugintools.py]: This is a strange way to get the addon ID, nevermind that "open with" is the correct way to do the file read.
Is there something I'm missing here like the addon is trying to read a different addon's settings?

File: plugin.video.fullycharged/default.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    plugintools.log("fullycharged.run")
+    
+    # Get params
+    params = plugintools.get_params()
+    
+    if params.get("action") is None:
+        main_list(params)
+    else:
+        action = params.get("action")
+        exec action+"(params)"
```
Comment (phil65) [inline plugin.video.fullycharged/default.py]: using exec is also far from secure, please get rid of it.

File: plugin.video.fullycharged/addon.xml (hunk 1 above)
Comment (zag2me) [inline plugin.video.fullycharged/addon.xml]: Thanks, yes Tamland suggested moving icon.png and fanart.png to the root and modifying the asset tag accordingly. This maintains backward compatibility. I'll change the xbmc.python to 2.14.0 and branch to Gotham.

File: plugin.video.fullycharged/plugintools.py (hunk 2 above)
Comment (zag2me) [inline plugin.video.fullycharged/plugintools.py]: plugin tools was written by someone else, I will need to check upstream otherwise not sure what to do.

**Pull request closed without merging**